.travis.yml

```diff
@@ -20,166 +17,144 @@ sudo: false
-               LC_CTYPE=C.ascii LC_ALL=C
-               PYTEST_VERSION=3.7
+          python: 3.5
+          env: TOXENV=py35-alldeps
+               TOXOPTS='--force-dep \'pytest==3.7\''
+               TOXARGS='--cov=astropy --remote-data=astropy'
+               CFLAGS='--coverage -fno-inline-functions -O0'
+               # TOXARGS='--remote-data=astropy --readonly'
+               # LC_CTYPE=C.ascii LC_ALL=C
+               # PIP_DEPENDENCIES="asdf"
 
         - os: linux
           stage: Initial tests
-          env: PYTHON_VERSION=3.7 CONDA_DEPENDENCIES=$CONDA_ALL_DEPENDENCIES
-               PIP_DEPENDENCIES="asdf"
-               SETUP_CMD='test -a "--durations=50"'
+          python: 3.7
+          dist: xenial
+          sudo: true
+          env: TOXENV=py37-alldeps
+               TOXARGS='--durations=50'
+               # PIP_DEPENDENCIES="asdf"
           compiler: clang
 
         # Full tests with coverage checks.
         - os: linux
-          env: SETUP_CMD='test --coverage --remote-data=astropy --readonly'
-               CONDA_DEPENDENCIES=$CONDA_ALL_DEPENDENCIES
-               PIP_DEPENDENCIES="codecov objgraph jplephem bintrees sortedcontainers asdf"
-               LC_CTYPE=C.ascii LC_ALL=C
+          python: 3.6
+          env: TOXENV=py36-alldeps
+               TOXARGS='--cov=astropy --remote-data=astropy'
                CFLAGS='--coverage -fno-inline-functions -O0'
-               MATPLOTLIB_VERSION=2.0
                EVENT_TYPE='push pull_request cron'
+               # TOXARGS='--remote-data=astropy --readonly'
+               # LC_CTYPE=C.ascii LC_ALL=C
+               # PIP_DEPENDENCIES="objgraph bintrees sortedcontainers asdf"
+               # MATPLOTLIB_VERSION=2.0
 
         # Try pre-release version of Numpy without optional dependencies
         - os: linux
-          env: NUMPY_VERSION=prerelease
+          python: 3.6
+          env: TOXENV=py36
+               TOXOPTS='--pre'
                EVENT_TYPE='push pull_request cron'
 
         # Do a PEP8/pyflakes test with flake8
         - os: linux
           stage: Initial tests
-          env: MAIN_CMD="flake8 astropy --count $FLAKE8_OPT" SETUP_CMD=''
+          python: 3.6
+          env: TOXENV=flake8
 
         # Try developer version of Numpy with optional dependencies and also
         # run all remote tests. Since both cases will be potentially
         # unstable, we combine them into a single unstable build that we can
         # mark as an allowed failure below.
-        - os: linux
-          env: NUMPY_VERSION=dev SETUP_CMD='test --remote-data'
-               CONDA_DEPENDENCIES=$CONDA_ALL_DEPENDENCIES
-               PIP_DEPENDENCIES="asdf"
-               MATPLOTLIB_VERSION=dev
+        # - os: linux
+        #   env: NUMPY_VERSION=dev SETUP_CMD='test --remote-data'
+        #        CONDA_DEPENDENCIES=$CONDA_ALL_DEPENDENCIES
+        #        PIP_DEPENDENCIES="asdf"
+        #        MATPLOTLIB_VERSION=dev
```

Comment:
I think we should add py??-dev builds in tox.ini which uses dev versions of a few dependencies (numpy and matplotlib for now). While I don't think we want configurations for all pinned versions in tox.ini, I think the dev space is a case it does make sense to include.

Reply:
How are these installed ? I don't know much about ci-helpers, so I was not sure of what really does `NUMPY_VERSION=dev`.

Reply:
Basically you can just pip install numpy and matplotlib from github using the ``git+https://www.github.com/...`` notation.

Reply:
Yes that's an option, is this what ci-helper does ?

Reply:
Yep!